ush/python/pygfs/task/aero_emissions.py

```diff
@@ -37,48 +44,439 @@ def __init__(self, config: Dict[str, Any]) -> None:
+        climfiles = np.sort(glob("{}{}".format(Config_dict.climfile_str, "*.nc")))
+        coarsen_scale = Config_dict.coarsen_scale
+        output_vars = Config_dict.output_vars
+        input_vars = Config_dict.input_vars
+
+        emission_map = {'qfed': 'qfed2.emis_*.nc4',
+                        'gbbepx': 'GBBEPx_all01GRID.emissions_v*.nc',
+                        'hfed': 'hfed.emis_*.x576_y361.*nc4'}
+
+        if emistype.lower() != 'blended':
+            try:
+                basefile = glob(emission_map[emistype.lower()])
+            except KeyError as err:
+                raise KeyError(f"FATAL ERROR: {emistype.lower()} is not a supported emission type, ABORT!")
+
+        if emistype.lower() == 'hfed':
+            AerosolEmissions.process_hfed(files=basefile, out_name=Config_dict.data_out['copy'][0][0], out_vars=output_vars, input_vars=input_vars)
+        else:
+            dset = AerosolEmissions.make_fire_emission(
+                d=current_date,
+                climos=climfiles,
+                ratio=ratio,
+                scale_climo=True,
+                coarsen_scale=coarsen_scale,
+                obsfile=basefile,
+                output_vars=output_vars,
+                input_vars=input_vars)
+
+            AerosolEmissions.write_ncf(dset, Config_dict.data_out['copy'][0][0])
+
     @staticmethod
     @logit(logger)
-    def execute(workdir: Union[str, os.PathLike], aprun_cmd: str) -> None:
-        """Run the executable (if any)
+    def process_hfed(files: list, out_name: str, out_vars: list = None, input_vars: list = None) -> None:
+        """
+        Process HFED files to generate fire emissions data.
+
+        Parameters:
+        -----------
+        files: list
+            List of HFED files to process.
+        out_name: str
+            Name of the output file to save the processed data.
+        out_vars: list, optional
+            List of output variables. Default is None.
+        input_vars: list, optional
+            List of input variables. Default is None.
+
+        Returns:
+        --------
+        None
+        """
+        vrs = out_vars
+        invars = input_vars
+
+        try:
+            if len(files) == 0:
+                raise Exception("FATAL ERROR: No HFED files were found")
+        except Exception as ee:
+            logger.exception("FATAL ERROR: HFED files not found.")
+            raise Exception(f"FATAL ERROR: Unable to find files, ABORT")
+
+        found_species = []
+        dset_dict = {}
+        for f in files:
+            index_good = [[i, v] for i, v in enumerate(invars) if v in f]
+            good = index_good[0][0]
+            found_species.append(index_good[0][1])
+            try:
+                logger.info("Opening HFED file: {filename}".format(filename=f))
+                with xr.open_dataset(f, decode_cf=False).biomass as da:
+                    da.name = vrs[good]
+                    dset_dict[vrs[good]] = da
```

Comment:
This bit with the enumerate is non-intuitive to parse. It looks like the purpose is to map input variables to output variables? If so, I would suggest replacing input_vars and out_vars with a dictionary with the mapping and removing this bit with enumerate.

```suggestion
        for filename in files:
            # Even better if the filenames are regular and we can replace this with a regex
            in_var = [var for var in in_vars if var in filename][0]
            out_var = variable_mapping[in_var]
            found_species.append(in_var)
            
            try:
                logger.info(f"Opening HFED file: {filename}")
                with xr.open_dataset(filename, decode_cf=False).biomass as da:
                    da.name = out_var
                    dset_dict[out_var] = da
```

Reply:
Attempted to address in 7ea3b689450, let me know what you think.